### PR TITLE
linux_pinebookpro_latest: Update patches for 5.13

### PR DESCRIPTION
**Pull request by zhaofengli**
https://github.com/nadiaholmquist/pbp-packages has been archived by the owner, so let's switch to Manjaro ARM's patchset instead. Note that Manjaro ARM dropped a few patches when they were updating to 5.13.

### CODE REVIEW

**Comment by samueldr** [comment]
cc @theotherjimmy

File: kernel/latest/default.nix [hunk 1]
```diff
@@ -49,19 +49,17 @@ linux_latest.override({
...
-    (nhp "0014-arm64-rockchip-add-DP-ALT-rockpro64.patch"                          "03k13jgcnz7wmks1y1fgzpjj2yvi114cbvprmnkyf8xrjns7x5q0")
-    (nhp "0015-ayufan-drm-rockchip-add-support-for-modeline-32MHz-e.patch"         "0z51whv0bjj45l5z3q4v0rqdvz62dh4qg8ccd87la9ga8y1v14cy")
-    (nhp "0021-usb-typec-bus-Catch-crash-due-to-partner-NULL-value.patch"          "0a4zd7ihd9pj6djgcj4ayaw7ff0xs9wqgmcvhwchwy766js3l5rp")
-    (nhp "0022-phy-rockchip-typec-Set-extcon-capabilities.patch"                   "0pqq856g0yndxvg9ipbx1jv6j4ldvapgzvxzvpirygc7f0wdrz49")
-    (nhp "0023-usb-typec-altmodes-displayport-Add-hacky-generic-altmode.patch"     "1vldwg3zwrx2ppqgbhc91l48nfmjkmwwdsyq6mq6f3l1cwfdn62q")
-    (nhp "0024-arm64-dts-rockchip-setup-USB-type-c-port-as-dual-dat.patch"         "0zwwyhryghafga36mgnazn6gk88m2rvs8ng5ykk4hhg9pi5bgzh9")
-    (nhp "0026-arm64-dts-rockchip-add-typec-extcon-hack.patch"                     "1kri47nkm6qgsqgkxzgy6iwhpajcx9xwd4rf8dldr6prb9f6iv3p")
-    (nhp "pbp-2d-fix.patch"                                                        "1hwd6clk1qnjyd4jl7kjn9pnilijz4brh1p5dnv8jzr2ajx2346j")
```
Comment (samueldr) [inline kernel/latest/default.nix:64]: I'm not sure what it did, but I assume this should still apply:

 - https://github.com/nadiaholmquist/pbp-packages/blob/master/linux-pbp/pbp-2d-fix.patch

We probably should try to get more info before just dropping it. I'm not sure what kind of perfs this workaround "fixes".

File: kernel/latest/default.nix [hunk 2]
```diff
@@ -49,19 +49,17 @@ linux_latest.override({
 
     # Misc. community patches
     # None are *required* for basic function.
-    # https://github.com/nadiaholmquist/pbp-packages/tree/master/linux-pbp
-    (nhp "0007-mmc-core-pwrseq_simple-disable-mmc-power-on-shutdown.patch"         "1d16gjgds670dzpkb8jjlymcpp1inab3mlzbpfdinrgvfy4pywhi")
```
Comment (samueldr) [inline kernel/latest/default.nix:53]: This was needed to fix an issue with Wi-Fi at reboot.

 - https://github.com/torvalds/linux/blob/master/drivers/mmc/core/pwrseq_simple.c

Upstream doesn't have an equivalent change, but that doesn't mean it wasn't fixed elsewhere.

(If there was a centralized distro-agnostic repository of information about vendor devices we wouldn't need to second-guess everything :()

File: kernel/latest/default.nix (hunk 1 above)
Comment (zhaofengli) [reply kernel/latest/default.nix:64]: Yeah, I tried searching as well and couldn't find much information. I haven't experienced any discernible difference in desktop with and without the patch.

**Comment by samueldr** [comment]
Additionally, while searching for some patches status, found this:

 - https://github.com/megous/linux/commits/pbp-5.13

We **do not** want ff2281b0a3463b110d06cb3e84685c60a6ae42cd as it would disable the Type-C DP output.

But the two others seem to be worthwhile.

One replaces `dts-rockchip-setup-USB-type-c-port-as-dual-data-role` while doing more, the other seems to not do anything useful, except be less noisy on a "defer".

**Comment by theotherjimmy** [comment]
It's working for me, and I can even add zfs support to the SD image, which was failing for 5.12

DP though USBC seems broken, with:
```
cdn-dp fec00000.dp: [drm:cmd_dp_checkk_sink_connection [rockchipdrm]] *ERROR* Get sink capability timed out
```

Though, I'll still be using this because I need the zfs support. Eventually getting DP out through UsbC would be cool too.

**Comment by theotherjimmy** [comment]
It's mysteriously failing to build on the pinebook-pro. That's disappointing. Though I cross-compiled with 5.13.2, and the pbp is building with 5.13.3

File: kernel/latest/default.nix (hunk 2 above)
Comment (fpletz) [reply kernel/latest/default.nix:53]: Not sure what the original issue was but wifi works after reboot for me without this patch.

**Comment by fpletz** [comment]
I've added some more manjaro patches which seemed applicable for the pinebook pro and added the two patches @samueldr mentioned over at my branch https://github.com/fpletz/wip-pinebook-pro/tree/linux-513-patches. Here is the commit https://github.com/fpletz/wip-pinebook-pro/commit/95a8644ef381b6149f31e58de6ed093a65ef67c0.

I've successfully built with 5.13.2 and booted successfully but I will have to do more testing, i.e. DP. Just thought I drop the current state of my work on this for @theotherjimmy.

**Comment by fpletz** [comment]
DP via USB-C also doesn't work for me. 

```
[    3.535060] rockchip-drm display-subsystem: bound fec00000.dp (ops cdn_dp_component_ops [rockchipdrm])
[    3.535196] cdn-dp fec00000.dp: Direct firmware load for rockchip/dptx.bin failed with error -2
[    4.549827] cdn-dp fec00000.dp: Direct firmware load for rockchip/dptx.bin failed with error -2
[    6.580565] cdn-dp fec00000.dp: [drm:cdn_dp_pd_event_work [rockchipdrm]] Not connected. Disabling cdn
[    6.580623] cdn-dp fec00000.dp: [drm:cdn_dp_pd_event_work [rockchipdrm]] Not connected. Disabling cdn
[    8.483187] cdn-dp fec00000.dp: [drm:cdn_dp_pd_event_work [rockchipdrm]] Connected, not enabled. Enabling cdn
[   13.681811] cdn-dp fec00000.dp: [drm:cdn_dp_check_sink_connection [rockchipdrm]] *ERROR* Get sink capability timed out
[   13.683091] cdn-dp fec00000.dp: [drm:cdn_dp_pd_event_work [rockchipdrm]] *ERROR* Enable dp failed -19
```

Edit: Current Manjaro ARM with 5.13.0 seems to have exactly the same problem.

**Comment by samueldr** [comment]
Right, so my initial assumption that DP support wasn't "cared for" in their update to 5.13 seems to hold true. Which means I was right in not looking into it earlier since I didn't have the time to *properly* port forward the changes :/.

**Comment by theotherjimmy** [comment]
My panfrost is broken with this, I'm trying applying https://patchwork.kernel.org/project/dri-devel/cover/20210630062751.2832545-1-boris.brezillon@collabora.com/ fingers crossed. EDIT; and it dose not apply cleanly. Now I'm giving fpletz's version.

**Comment by theotherjimmy** [comment]
I'm not seeing any change from fpletz's version. Though it seems that my systemd-based setup may be causing issues, as i3 launched from within systemd fails the same way.

**Comment by samueldr** [comment]
And compared to 5.12?

**Comment by zhaofengli** [comment]
I've cherry-picked @fpletz's commit and updated the PR.

**Comment by theotherjimmy** [comment]
samueldr, I have no idea what 5.12 did, I does not boot my ZFS-based system (the zfs was only compatible with < 5.10, making it unbuildable)

**Comment by plabadens** [comment]
Hmm, I'm having a similar issue as @theotherjimmy: sway only works if I enable software rendering, so there must be something wrong with panfrost on 5.13, or at least some incompatibility with my existing config.

**Comment by zhaofengli** [comment]
> Hmm, I'm having a similar issue as @theotherjimmy: sway only works if I enable software rendering, so there must be something wrong with panfrost on 5.13, or at least some incompatibility with my existing config. 

I'm using the git version of Sway from the [nixpkgs-wayland](https://github.com/nix-community/nixpkgs-wayland) overlay. However, the entire system occasionally freezes and I'm not getting useful logs. I tested the [git version of Mesa](https://github.com/zhaofengli/nixpkgs/commit/ccc5c6f86c39da2ba22e2b3cb729f3c61f34b21b) but it didn't appear to improve stability for me.

**Comment by jbaum98** [comment]
I'm not sure this is helpful, but based on this thread and the [Manjaro PKGBUILD file](https://gitlab.manjaro.org/manjaro-arm/packages/core/linux/-/blob/5e308f6f7d53e12a7ab7e59d819eb2f71410504d/PKGBUILD), I got 5.14.13 working on my pinebook pro. The only issue is the suspend is broken again, but I think that only worked for me on 5.11 (so it doesn't work on 5.10 either). [Here](https://git.sr.ht/~jakewaksbaum/pi/tree/manjaro_kernel/item/kernel/default.nix) is my nix expression for the kernel.

The config Manjaro provides works fine, except that it compresses modules with `gz` where NixOS seems to need them to be compressed with `xz` since its version of `kmod` doesn't support `gz`.

**Comment by samueldr** [comment]
Hi, thanks for the contribution, and effort.

I've decided to keep it simple, and just drop any customized kernel, if I wanted to move forward.

It'd probably still make sense for people to coordinate on patch sets to use, but as long as you don't need an external display, I don't think any patches are needed. I'm not sure what would be a good way for coordinating such efforts though.

So uh, good news, it mostly works just fine with mainline.

**Pull request closed without merging**